obelix/idefix/services/google/client.go

```diff
@@ -0,0 +1,70 @@
+package google
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"google.golang.org/api/calendar/v3"
+	"google.golang.org/api/option"
+	"google.golang.org/protobuf/types/known/structpb"
+	"idefix/devauth"
+	"idefix/operator"
+	"idefix/services/github"
+	"log"
+	"net/http"
+)
+
+// Client represents a minimal Google client able to make OAuth2.0 authenticated requests.
+type Client struct {
+	Requester *http.Client
+	Scope     []string
+	Operator  *operator.IdefixOperator
+}
+
+// configure get credentials to connect to Google API
+func (c *Client) configure() error {
+	//TODO Credentials API
+	client, err := devauth.New(context.Background(), c.Scope)
+
+	if err != nil {
+		return fmt.Errorf("failed to configure google client")
+	}
+
+	c.Requester = client.Clt
+
+	return nil
+}
+
+// CalendarNearestEvent check if event's calendar begin before 10 minutes
+func (c *Client) CalendarNearestEvent(taskID string, prm *structpb.Struct) error {
+	var gc GCalendar
+
+	if err := c.configure(); err != nil {
+		return fmt.Errorf("failed to configure github client: %w", err)
+	}
+
+	srv, err := calendar.NewService(context.Background(), option.WithHTTPClient(c.Requester))
+	if err != nil {
+		log.Fatalf("Unable to retrieve Calendar client: %v", err)
```

Comment:
Are you sure we want to exit program if we can't use one of our action?

Reply:
Nice catch ! Removed in https://github.com/Cleopha/ifttt-like/pull/55/commits/f4ec45d8b036b032e2eb1b7c4341eac9b18706f8

Reply:
that's my boy

Reply:
@PtitLuca please check the linter